bundle-types/action.yml

```diff
@@ -0,0 +1,120 @@
+name: "Grafana Bundle Types"
+description: "Bundles a Grafana plugins types and pushes them to npm."
+
+outputs:
+  typesFilePath:
+    description: "The path to the built types file."
+    value: ${{ steps.set-outputs.outputs.typesFilePath }}
+
+inputs:
+  entry-point:
+    description: "Location of types file to bundle."
+    required: false
+    default: "./src/types/index.ts"
+  ts-config:
+    description: "A path to the tsconfig file to use when bundling types."
+    required: false
+    default: ""
+  node-version:
+    description: "Version of node"
+    required: false
+    default: "20"
+
+runs:
+  using: "composite"
+  steps:
+    - uses: actions/setup-node@v4
+      with:
+        node-version: "${{ inputs.node-version }}"
+
+    - name: Install dependencies
+      run: ${{ github.action_path }}/pm.sh install
+      shell: bash
+
+    - name: Get plugin id and version
+      id: metadata
+      run: |
+        PLUGIN_ID=$(cat src/plugin.json | jq -r .id)
+        echo "plugin-id=${PLUGIN_ID}" >> $GITHUB_OUTPUT
+        PLUGIN_VERSION=$(cat package.json | jq -r .version)
+        echo "plugin-version=${PLUGIN_VERSION}" >> $GITHUB_OUTPUT
```

Comment:
Hmm... plugins that live in repos alongside other code (e.g. monorepos) won't work with this. Should we concern ourselves with this now? Maybe we ask for the plugin-id and version to be input by the consumer? 🤔

Reply:
that's a good idea, we can go for optional parameters that default to read the values from here?

Reply:
Added a couple of extra inputs in https://github.com/grafana/plugin-actions/pull/40/commits/dc51c2bf7b915fc52bdc23d6c12f1de758ebf029 to cater for these paths.